admin/lib/solidus_admin/container.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require "dry/system"
+require "dry/system/container"
+require "dry/system/component"
+require "view_component"
+require "solidus_admin/system/loaders/host_overridable_constant"
+
+module SolidusAdmin
+  # Global registry for host-injectable components.
+  #
+  # We use this container to register all the components that can be
+  # overridden by the host application.
+  #
+  # @api private
+  class Container < Dry::System::Container
+    configure do |config|
+      config.root = Pathname(__FILE__).dirname.join("../..").realpath
+      config.component_dirs.add("app/components") do |dir|
+        dir.loader = System::Loaders::HostOverridableConstant.method(:call).curry["components"]
+        dir.namespaces.add "solidus_admin", key: nil
+      end
+    end
```

Comment:
Hey @elia, as pointed out in a [previous comment](https://github.com/solidusio/solidus/pull/5099#discussion_r1210069301), there's definitely nothing we couldn't build on our own. However, for the sake of this spike, I wanted to focus on the architecture and the interface to interact with it instead of getting lost in implementation details. dry-system allowed me to do that, but I have no problems revisiting it and evaluating if it's enough to build something like a Hash-based registry.

I'm not sure which direction we're going to take on other topics, like service objects on the new admin, but having dry-system in place can help us with moving and changing things faster, as its declarative configuration is very flexible. Then, it can also give us other valuable features like logging, monitoring, or testing support (stubbing a component while testing). We can assess how much of it we end up using, and then decide whether to go with or without it. What do you think?

Reply:
Maybe it's unfamiliarity with DRY, but looks like for our case it's an additional layer of injections on top of the already sufficient initializer params, which I also find more explicit.

I think having a more complete context might help in appreciating pros and cons of each solution, thoughts on holding a little bit on merging and keeping this PR as a reference maybe coming back to it once we start implementing? 

_I think we can merge right away the first commits setting up the Base component class_

Reply:
After talking offline, we decided to merge everything now and re-evaluate along the way :+1: